perllib/FixMyStreet/DB/Result/Problem.pm

```diff
@@ -606,21 +606,33 @@ sub meta_line {
             $meta = sprintf( _('Reported anonymously at %s'), $date_time );
         }
     } else {
-        if ( $problem->service and $category && $category ne _('Other') ) {
-            $meta = sprintf(
-                _('Reported via %s in the %s category by %s at %s'),
-                $problem->service, $category,
-                $problem->name,    $date_time
-            );
-        } elsif ( $problem->service ) {
-            $meta = sprintf( _('Reported via %s by %s at %s'),
-                $problem->service, $problem->name, $date_time );
-        } elsif ( $category and $category ne _('Other') ) {
-            $meta = sprintf( _('Reported in the %s category by %s at %s'),
-                $category, $problem->name, $date_time );
+      my $problem_name;
+
+      if ( $c->user and $c->user->has_permission_to('view_body_contribute_details', $problem->bodies_str_ids) ) {
```

Comment:
You can put the other if on the line below in with this if as both the else cases below do the same thing, iyswim.

Also, just as a comment – you can also use `$c->user_exists` which then doesn't touch the db, if you only need to know if someone is logged in, for example.